lighthouse-core/gather/gatherers/accessibility.js

```diff
@@ -27,10 +27,12 @@ const axe = fs.readFileSync(
 // This is run in the page, not Lighthouse itself.
 /* istanbul ignore next */
 function runA11yChecks() {
-  axe.a11yCheck(document, function(results) {
-    // __returnResults is magically inserted by driver.evaluateAsync
-    __returnResults(results);
+  var fulfill;
+  var result = new Promise(f => {
+    fulfill = f;
   });
+  axe.a11yCheck(document, fulfill);
+  return result;
```

Comment:
rather than expose the `resolve` function external to the promise, you could instead do
```js
function runA11yChecks() {
  return new Promise((resolve, reject) => {
    axe.a11yCheck(document, resolve);
  });
}
```